src/GitUtils.ps1

```diff
@@ -415,6 +415,52 @@ function Get-SshAgent() {
     return 0
 }
 
+function Get-NativeSshAgent {
+    # The ssh.exe binary version must include "OpenSSH"
+    # The windows ssh-agent service must exist
+    $service = Get-Service ssh-agent
+    $executableMatches = Get-Command ssh.exe | ForEach-Object FileVersionInfo | Where-Object ProductVersion -match OpenSSH
+    $valid = $service -and $executableMatches
+
+    if ($valid) {
+        return $service;
+    }
+
+    return $false;
+}
+function Start-NativeSshAgent([switch]$Quiet) {
+    $service = Get-NativeSshAgent
+    
+    if (!$service) {
+        return $false;
+    }
+
+    # Enable the servivce if it's disabled and we're an admin
+    if ($service.StartupType -eq "Disabled") {
+        if (Test-Administrator) {
+            Set-Service $service -StartupType Automatic
```

Comment:
IMO `Manual` seems maybe more appropriate here if they're using a manual `Start-SshAgent` function.

Reply:
There are going to be times when you want to start VS after a reboot and before you've fired up a PowerShell console.  In those cases, using Visual Studio against a Git repo using an ssh URL will be no bueno because Windows did not automatically start ssh-agent.

Reply:
Also another command that probably should be Windows only.  BTW, I do this test with:
```
if (($PSVersionTable.PSVersion.Major -lt 6) -or $IsWindows) {
```
That seems to work with strict mode.

Reply:
`Manual` startup would indeed be less presumptuous but I just don't see folks using SSH keys w/passphrases on Windows as wanting the startup mode to be anything other than `Automatic`.   At the very least, a switch parameter for automatic startup would be nice.

Reply:
As long as `Get-NativeSshAgent` returns falsey on non-Windows, no changes should be necessary here.

Let's go with Manual, plus an option if @JeremySkinner is up for it. Priority is to stop it being completely broken.

Reply:
- Added `StartupType` parameter to the function, which defaults to manual, which is now passed through from `Start-SshAgent`
- Version check is now handled in `Get-NativeSshAgent` so there shouldn't be a need for another one in `Start-SshAgent`

I also realised the `$service.StartupType` property is only defined in PS Core, so switched to `$service.StartType` instead as this is available in PS 4 and newer. I'm assuming we don't care about older PS versions as they won't run on win10 anyway.